scraper/src/fetch_german_names.js

```diff
@@ -0,0 +1,212 @@
+    return;
+  }
+
+  const entities = data.entities;
+
+  if (entities[-1]) {
+    // If there is a -1, the site doesn't exist
+    return;
+  }
+
+  const keys = Object.keys(entities);
+  const entity = entities[keys[0]];
+
+  const label_entity = entity.labels;
+  if (label_entity && label_entity.de) {
+    germanNames.push(label_entity.de.value);
+  } else {
+    //If there is no label for the plant, then there are no other relevant german common names.
+    return;
+  }
+
+  const dewiki_entity = entity.sitelinks.dewiki;
+  if (dewiki_entity) {
+    germanNames.push(dewiki_entity.title);
+  }
+};
+
+/**
+ * Fetches the German name of the plant from the Wikidata API.
+ * Sets the 'common_name_de' property of every plant in the array.
+ * Uses the current entry in 'common_name_de' and extends it with data found on Wikidata.
+ *
+ * @param {Array} plants - An array containing a part of all plant objects.
+ */
+const processPlants = async (plants) => {
+  for (const plant of plants) {
+    const unique_name = plant["unique_name"];
+    const germanNames = [];
+
+    if (plant.common_name_de && plant.common_name_de !== "true") {
+      const existingGermanNames = plant.common_name_de.split(",");
+      existingGermanNames.forEach((existingGermanName) => {
+        germanNames.push(existingGermanName);
+      });
+    }
+
+    plant.common_name_de = null;
+
+    await axios
+      .get(
+        `https://www.wikidata.org/w/api.php?action=wbgetentities&sites=enwiki&titles=${unique_name}&normalize=&languages=de&format=json`
+      )
+      .then((response) => {
+        // log response code
+        handleResponseAddGermanNamesFound(response, germanNames);
+      })
+      .catch((error) => {
+        console.error(
+          `[ERROR] Could not get German names for "${unique_name}" from Wikidata: `,
+          error.message
+        );
+      })
+      .finally(() => {
+        if (germanNames.length === 0) {
+          return;
+        }
+        const cleanedGermanNames = filterGermanNames(germanNames, unique_name);
+
+        if (cleanedGermanNames.length > 0) {
+          GermanNamesFound++;
+          plant["common_name_de"] = cleanedGermanNames.join(", ");
+        }
+      });
+  }
+};
+
+/**
+ * Sets the 'common_name_de' property of every plant in the array.
+ * Splits up the entire plants array into smaller arrays to improve performance for name fetching.
+ *
+ * @param {Array} plants - An array containing all plant objects.
+ */
+const fetchGermanNamesForPlantsConcurrent = async (plants) => {
+  const MAX_CONCURRENT_REQUESTS = 25;
+
+  console.log("[INFO] Fetching German common names ...");
+
+  // Chunk the plants into batches of MAX_CONCURRENT_REQUESTS
+  const chunks = [];
+  const chunk_size = plants.length / MAX_CONCURRENT_REQUESTS + 1;
+  for (let i = 0; i < plants.length; i += chunk_size) {
+    chunks.push(plants.slice(i, i + chunk_size));
+  }
+
+  // Process each chunk concurrently using Promise.all
```

Comment:
in my opinion this comment is not necessary since the code itself is meaningful enough